### PR TITLE
Nginx: Ignore all internal-api paths

### PR DESCRIPTION
**Pull request by junderw**
TODO: Cleanup these new intensive endpoints that have been added recently.

- [ ] Remove the endpoints from the backend so they are not exposed to the frontend.
- [ ] Ensure that only the backend can access the mempool-electrs.

### CODE REVIEW

**Comment by junderw** [comment]
@mononaut Please assist if you have time.

@wiz Nginx is not my forte, if you have any comments I'd appreciate it.

Since we only need a lot of these new mempool-electrs endpoints for the backend, we should only allow the backend to access these endpoints.

File: backend/src/api/bitcoin/esplora-api.ts [hunk 1]
```diff
@@ -214,11 +214,11 @@ class ElectrsApi implements AbstractBitcoinApi {
   }
 
   async $getMempoolTransactions(txids: string[]): Promise<IEsploraApi.Transaction[]> {
-    return this.failoverRouter.$post<IEsploraApi.Transaction[]>('/mempool/txs', txids, 'json');
+    return this.failoverRouter.$post<IEsploraApi.Transaction[]>('/internal-api/mempool/txs', txids, 'json');
```
Comment (wiz) [inline backend/src/api/bitcoin/esplora-api.ts:217]:
```suggestion
    return this.failoverRouter.$post<IEsploraApi.Transaction[]>('/internal/mempool/txs', txids, 'json');
```

File: backend/src/api/bitcoin/esplora-api.ts [hunk 2]
```diff
@@ -214,11 +214,11 @@ class ElectrsApi implements AbstractBitcoinApi {
...
   }
 
   async $getAllMempoolTransactions(lastSeenTxid?: string): Promise<IEsploraApi.Transaction[]> {
-    return this.failoverRouter.$get<IEsploraApi.Transaction[]>('/mempool/txs' + (lastSeenTxid ? '/' + lastSeenTxid : ''));
+    return this.failoverRouter.$get<IEsploraApi.Transaction[]>('/internal-api/mempool/txs' + (lastSeenTxid ? '/' + lastSeenTxid : ''));
```
Comment (wiz) [inline backend/src/api/bitcoin/esplora-api.ts:221]:
```suggestion
    return this.failoverRouter.$get<IEsploraApi.Transaction[]>('/internal/mempool/txs' + (lastSeenTxid ? '/' + lastSeenTxid : ''));
```

File: nginx-mempool.conf [hunk 3]
```diff
@@ -39,6 +39,11 @@
 		try_files $uri $uri/ /$1/index.html =404;
 	}
 
+	# any path containing .*/internal-api/.* anywhere is ignored
+	location ~ ^/.*?/internal-api/ {
+		return 404;
+	}
+
```
Comment (wiz) [inline nginx-mempool.conf:46]:
```suggestion
```

File: production/nginx/location-api.conf [hunk 4]
```diff
@@ -2,6 +2,11 @@
 # mempool #
 ###########
 
+# any path containing .*/internal-api/.* anywhere is ignored
+location ~ ^/.*?/internal-api/ {
+	return 404;
+}
+
```
Comment (wiz) [inline production/nginx/location-api.conf:9]:
```suggestion
location /api/internal/ {
	return 404;
}
location /api/v1/internal/ {
	return 404;
}
```

**Review by wiz — CHANGES_REQUESTED** [review]
NACK for URL schema and nginx configuration, see suggested changes and also please add i.e. `/testnet/api/internal` and `/testnet/api/v1/internal` in `location-testnet-api.conf` and similar for signet, liquid, liquidtestnet, and bisq routing as well

**Comment by junderw** [comment]
Merged suggestions and added 404 endpoints to each conf file with their respective root paths.

**Comment by mononaut** [comment]
rebased on current master [1e69f3b3]

**Review by wiz — APPROVED** [review]
Tested ACK @ v3.0.0-dev [ee90a96d2]

**Pull request merged**